miniapps/nurbs/nurbs_patch_ex1.cpp

```diff
@@ -0,0 +1,340 @@
+//               MFEM Example 1 - NURBS with patch-wise assembly
+//
+// Compile with: make nurbs_patch_ex1
+//
+// Sample runs:  nurbs_patch_ex1 -incdeg 3 -ref 2 -iro 8 -patcha
+//               nurbs_patch_ex1 -incdeg 3 -ref 2 -iro 8 -patcha -pa
+//               nurbs_patch_ex1 -incdeg 3 -ref 2 -iro 8 -patcha -fint
+//
+// Description:  This example code demonstrates the use of MFEM to define a
+//               simple finite element discretization of the Laplace problem
+//               -Delta u = 1 with homogeneous Dirichlet boundary conditions.
+//               Specifically, we discretize using a FE space of the specified
+//               order, or if order < 1 using an isoparametric/isogeometric
+//               space (i.e. quadratic for quadratic curvilinear mesh, NURBS for
+//               NURBS mesh, etc.)
+//
+//               This example is a specialization of ex1 which demonstrates
+//               patch-wise matrix assembly and partial assembly on NURBS
+//               meshes. There is the option to compare run times of patch
+//               and element assembly, as well as relative error computation.
+
+#include "mfem.hpp"
+#include <fstream>
+#include <iostream>
+
+using namespace std;
+using namespace mfem;
+
+void AssembleAndSolve(LinearForm & b, BilinearFormIntegrator * bfi,
+                      Array<int> const& ess_tdof_list,
+                      const bool pa, const bool static_cond,
+                      const bool algebraic_ceed, GridFunction & x);
+
+int main(int argc, char *argv[])
+{
+   // 1. Parse command-line options.
+   const char *mesh_file = "../../data/beam-hex-nurbs.mesh";
+   int order = -1;
+   bool static_cond = false;
+   bool pa = false;
+   const char *device_config = "cpu";
+   bool visualization = true;
+   bool algebraic_ceed = false;
+   bool patchAssembly = false;
+   bool reducedIntegration = true;
+   bool compareToElementWise = true;
+   int nurbs_degree_increase = 0;  // Elevate the NURBS mesh degree by this
+   int ref_levels = 0;
+   int ir_order = -1;
+
+   OptionsParser args(argc, argv);
+   args.AddOption(&mesh_file, "-m", "--mesh",
+                  "Mesh file to use.");
+   args.AddOption(&static_cond, "-sc", "--static-condensation", "-no-sc",
+                  "--no-static-condensation", "Enable static condensation.");
+   args.AddOption(&pa, "-pa", "--partial-assembly", "-no-pa",
```

Comment:
`-pa` with `-no-patcha` would segfault. Maybe we should add a check!

Reply:
Added a check.